Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt

```diff
@@ -17,38 +17,40 @@
 package com.example.owl.ui
 
 import androidx.activity.OnBackPressedDispatcher
-import androidx.compose.animation.Crossfade
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.Providers
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.savedinstancestate.rememberSavedInstanceState
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import com.example.owl.ui.course.CourseDetails
 import com.example.owl.ui.courses.Courses
 import com.example.owl.ui.onboarding.Onboarding
 import com.example.owl.ui.utils.AmbientBackDispatcher
-import com.example.owl.ui.utils.Navigator
 import com.example.owl.ui.utils.ProvideDisplayInsets
 import com.example.owl.ui.utils.ProvideImageLoader
 
 @Composable
 fun OwlApp(backDispatcher: OnBackPressedDispatcher) {
-    @Suppress("RemoveExplicitTypeArguments")
-    val navigator: Navigator<Destination> = rememberSavedInstanceState(
-        saver = Navigator.saver<Destination>(backDispatcher)
-    ) {
-        Navigator(Destination.Onboarding, backDispatcher)
-    }
-    val actions = remember(navigator) { Actions(navigator) }
 
     Providers(AmbientBackDispatcher provides backDispatcher) {
         ProvideDisplayInsets {
             ProvideImageLoader {
-                Crossfade(navigator.current) { destination ->
-                    when (destination) {
-                        Destination.Onboarding -> Onboarding(actions.onboardingComplete)
-                        Destination.Courses -> Courses(actions.selectCourse)
-                        is Destination.Course -> CourseDetails(
-                            destination.courseId,
+                val navController = rememberNavController()
+
+                val actions = remember(navController) { Actions(navController) }
+                NavHost(navController = navController, startDestination = Destinations.Onboarding) {
+                    composable(Destinations.Onboarding) {
+                        Onboarding(actions.onboardingComplete)
+                    }
+                    composable(Destinations.Courses) {
+                        Courses(actions.selectCourse)
+                    }
+                    composable(Destinations.CourseDetails + "/{courseId}") { backStackEntry ->
```

Comment:
The argument string `courseId` is reused twice. Would it make sense to extract this into the NavGraph class somehow?

Reply:
What do you think about this? A little verbose and a lot of curly braces

Reply:
Better. I'll create something similar for Jetsnack!